src/components/Modal.tsx

```diff
@@ -0,0 +1,34 @@
+import React from "react"
+import {ModalProps} from "../interface/ModalProps.ts";
+import {CardListType} from "../enums/CardListType.ts";
+import {CardModalType} from "../type/CardModalType.ts";
+
+const Modal: React.FC<ModalProps> = ({isOpen, selectCard}) => {
+    const cardSelect = (cardName: string) => {
+        selectCard(cardName)
+    }
+    const cardModalArray = (): CardModalType[] => {
+        let idCounter = 1
+        return Object.values(CardListType).map((cardType) => ({
+            id: idCounter++,
+            cardListType: cardType
+        }))
+    }
+
+    const cardModalTypes = cardModalArray();
+
+    return (
+        <div className="modal-dimmed" style={{display: isOpen ? "flex" : "none"}}>
+            <div className="modal">
+                {cardModalTypes.map((card) => (
+                    <div key={card.id} className="modal-item-container" onClick={() => cardSelect(card.cardListType)}>
```

Comment:
c: cardType 자체가 id로 기능해 따로 id를 만들 필요는 없어 보입니다.

```suggestion
                // cardModalTypes는 cardTypes를 배열로 변환한 값으로 수정
                {cardModalTypes.map((cardType) => (
                    <div key={cardType} className="modal-item-container" onClick={() => cardSelect(cardType)}>
```